### PR TITLE
Coherent units in `TemporalModel.sample_time`

### PR DESCRIPTION
**Pull request by fabiopintore**
This is a very minimal fix to preserve units coherence in the `TemporalModel.sample_time`. 

The time sampling is indeed performed over a number of time bins which is calculated internally on the basis of the `ontime` and the `t_delta`. 
In the current version, the `ontime` is calculated in seconds but it takes the units from `t_delta`. The latter is by default in seconds, but it can be passed by the user. 
If the units of `ontime` and `t_delta` are not consistent, this introduces an issue.

### CODE REVIEW

**Review by bkhelifi — APPROVED** [review]

File: gammapy/modeling/models/temporal.py [hunk 1]
```diff
@@ -198,7 +198,7 @@ def sample_time(self, n_events, t_min, t_max, t_delta="1 s", random_state=0):
         t_delta = u.Quantity(t_delta)
         random_state = get_random_state(random_state)
 
-        ontime = u.Quantity((t_max - t_min).sec, t_delta.unit)
+        ontime = u.Quantity((t_max - t_min).sec, t_delta.to("s").unit)
```
Comment (registerrier) [inline gammapy/modeling/models/temporal.py:201]:
```suggestion
        ontime = (t_max - t_min).to("s")
```

**Review by registerrier — CHANGES_REQUESTED** [review]
Thanks @fabiopintore . I think it would be simpler to convert the `TimeDelta` to a Quantity in seconds.

**Review by cgalelli — APPROVED** [review]
Thanks @fabiopintore . An alternative solution could be to use .decompose() but this is completely fine (and probably more efficient)

**Review by QRemy — APPROVED** [review]
Thanks @fabiopintore, no comment from my side.

**Comment by QRemy** [comment]
Backport to 1.0.x is not needed the code has correct a unit `ontime = u.Quantity((t_max - t_min).sec, "s")`

**Pull request merged**